tests/unit/test_charm.py

```diff
@@ -18,129 +19,152 @@
-        },
-    )
-
-
-def check_exec_args(harness, return_value, *args, **kwargs):
-    assert args == (
-        [
-            "lego",
-            "--email",
-            harness._charm._email,
-            "--accept-tos",
-            "--csr",
-            "/tmp/csr.pem",
-            "--server",
-            harness._charm._server,
-            "--dns",
-            harness._charm._plugin,
-            "run",
-        ],
-    )
-    assert kwargs == {
-        "timeout": 300,
-        "working_dir": "/tmp",
-        "environment": harness._charm._plugin_configs,
-        "combine_stderr": False,
-        "encoding": "utf-8",
-        "group": None,
-        "group_id": None,
-        "user": None,
-        "user_id": None,
-        "stderr": None,
-        "stdin": None,
-        "stdout": None,
-    }
-
-    return return_value
-
-
-def test_request(harness):
-    harness._backend._pebble_clients["lego"].exec = partial(
-        check_exec_args, harness, Mock(wait_output=lambda: (None, None))
-    )
-    harness._backend._pebble_clients["lego"].push(
-        "/tmp/.lego/certificates/foo.crt", source=test_lego.read_bytes(), make_dirs=True
+test_cert = Path(__file__).parent / "test_lego.crt"
+TLS_LIB_PATH = "charms.tls_certificates_interface.v1.tls_certificates"
+
+
+class AcmeTestCharm(AcmeClient):
+    def __init__(self, *args):
+        """Uses the Orc8rBase library to manage events."""
+        super().__init__(*args)
+        self._server = "https://acme-staging-v02.api.letsencrypt.org/directory"
+
+    @property
+    def _email(self) -> str:
+        return "example@email.com"
+
+    @property
+    def _plugin(self) -> str:
+        return "namecheap"
+
+    @property
+    def _plugin_config(self):
+        return None
+
+
+class TestCharm(unittest.TestCase):
+    def setUp(self):
+        self.harness = Harness(
+            AcmeTestCharm,
+            meta=yaml.safe_dump(
+                {
+                    "name": "lego",
+                    "containers": {"lego": {"resource": "lego-image"}},
+                    "provides": {"certificates": {"interface": "tls-certificates"}},
+                }
+            ),
+        )
+
+        self.harness.set_leader(True)
+        self.harness.set_can_connect("lego", True)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+        self.r_id = self.harness.add_relation("certificates", "remote")
+        self.harness.add_relation_unit(self.r_id, "remote/0")
+
+    def test_given_empty_pebble_plan_when_pebble_ready_then_status_is_active(self):
+        initial_plan = self.harness.get_container_pebble_plan("lego")
+        self.assertEqual(initial_plan.to_yaml(), "{}\n")
+        expected_plan = {}
+        container = self.harness.model.unit.get_container("lego")
+        self.harness.charm.on.lego_pebble_ready.emit(container)
+        updated_plan = self.harness.get_container_pebble_plan("lego").to_dict()
+        self.assertEqual(expected_plan, updated_plan)
```

Comment:
```suggestion
```
Since pebble_ready doesn't set the pebble plan I wouldn't validate this behaviour here.

Reply:
✅